### PR TITLE
(maint) Update the Forge API base url

### PR DESCRIPTION
**Pull request by binford2k**
The Forge API is now primarily served from the `puppet.com` domain. DNS
still resolves from the old domain, but the legacy `*.puppetlabs.com`
certificate [has been removed](https://headwayapp.co/puppet-forge-updates/removed-legacy-*-puppetlabs-com-ssl-certificate-143941).
For most users, this will have no effect because the main `puppet.com`
certificate includes `puppetlabs.com` as a SAN, but in some outdated or
otherwise broken SSL stacks, this might prevent SSL validation.

To remove that edge case, this PR updates the baseurl to the current
domain name.

### CODE REVIEW

**Comment by mwaggett** [comment]
@binford2k Is this still something we should do? We ended up needing to revert this change from the 2.6.x stream because it broke something in CI. Wondering if we should try again or just close this.

**Comment by binford2k** [comment]
🤷‍♀️ 

At some point yes, we totally should. But it's clearly less of a problem than I thought it would be, so it's low priority.

File: CHANGELOG.mkd [hunk 1]
```diff
@@ -7,7 +7,8 @@ Unreleased
 - Add 'strip\_component' environment source configuration setting, to allow deploying Git branches named like "env/production" as Puppet environments named like "production". [#1128](https://github.com/puppetlabs/r10k/pull/1128)
 - A warning will be emitted when the user supplies conflicting arguments to module definitions in a Puppetfile, such as when specifying both :commit and :branch [#1130](https://github.com/puppetlabs/r10k/pull/1130)
 - Add optional standard module and environment specification interface: name, type, source, version. These options can be used when specifying environments and/or modules in a yaml/exec source, as well as when specifying modules in a Puppetfile. Providing the standard interface simplifies integrations with external services [#1131](https://github.com/puppetlabs/r10k/pull/1131)
-
```
Comment (mwaggett) [inline CHANGELOG.mkd:10]: (nit) don't need this newline

**Review by mwaggett — COMMENTED** [review]
so yeah I'm hoping to get a chance to try this again soon. I forget exactly which tests broke when we tried merging this into an older branch, but I'll poke around (it was something in PE tests).

**Comment by mwaggett** [comment]
would be good to squash these commits though

**Comment by binford2k** [comment]
@mwaggett is the CI breakage still a thing?

**Comment by mwaggett** [comment]
@binford2k yes. hoping to get a chance to look into it soon.

**Comment by mwaggett** [comment]
@binford2k Were you ever able to get Forge folks to look into the CI breakage? It's (at least some of) the integration tests you updated in this PR that fail when you change the url from `puppetlabs` to `puppet`.

**Comment by caseywilliams** [comment]
Adding a bit of a summary here:

This breaks PE integration tests, we believe because of the use of [beaker-puppet's deprecated  `stub_forge_on` method](https://github.com/voxpupuli/beaker-puppet/blob/376a735afee0735dc7e7547310a060a738d6f5da/lib/beaker-puppet/helpers/puppet_helpers.rb#L652), which [was removed recently](https://github.com/voxpupuli/beaker-puppet/pull/160) - this won't be fixed in a future release of beaker-puppet. IMO, these tests should be either refactored or at least temporarily disabled if we want to get this change merged.

**Comment by mwaggett** [comment]
*The [PR](https://github.com/voxpupuli/beaker-puppet/pull/160) to remove `stub_forge_on` has not been merged yet, fwiw

I'm extremely hesitant to "temporarily" disable because there's a 95% chance that that means "permanently". If we don't think those tests are providing value, I'm happy to get rid of them, but we should make that decision consciously. 
As for refactoring, were you just thinking of using the generic `stub_hosts_on` method or did you have something else in mind?

**Comment by caseywilliams** [comment]
I don't have any strong opinions about how to solve this (and I'm not familiar with this codebase myself), so however the PE teams want to address this works for me. I asked around and I'm not aware of any plans to fully remove `forgeapi.puppetlabs.com` anytime soon, so whenever works for you is fine. `forgeapi.puppet.com` _is_ the official URL, though, so it should eventually become the default in r10k.

**Pull request closed without merging**